### PR TITLE
Fix SEGV in provider and support special task token.

### PR DESCRIPTION
**Pull request by jortel**
Fix SEGV in provider when token is invalid.
Add support for special task token.

### CODE REVIEW

File: api/base.go [hunk 1]
```diff
@@ -257,14 +259,12 @@ func (h *BaseHandler) modBody(
...
+// CurrentUser gets username from Keycloak auth token.
 func (h *BaseHandler) CurrentUser(ctx *gin.Context) (user string) {
 	token := ctx.GetHeader("Authorization")
-	// Consider later add username to ctx and cache for single request?
-	user, err := h.AuthProvider.User(token)
+	user, err := auth.CurrentUser(h.AuthProvider, token)
 	if err != nil {
-		fmt.Printf("Failed to get current user, err: %v\n", err)
-		user = ""
+		Log.Error(err, "Failed to get current user.")
```
Comment (jortel) [inline api/base.go:267]: Let's log this instead.

File: auth/provider.go [hunk 2]
```diff
@@ -39,7 +39,8 @@ func (r *NoAuth) Scopes(token string) (scopes []Scope, err error) {
 //
 // User mocks username for NoAuth
 func (r *NoAuth) User(token string) (name string, err error) {
-	return "admin.noauth", nil
+	name = "admin.noauth"
```
Comment (jortel) [inline auth/provider.go:42]: More consistent when using named return values.

File: auth/provider.go [hunk 3]
```diff
@@ -127,13 +128,13 @@ func (r *Keycloak) newScope(s string) (scope KeycloakScope) {
...
 	defer cancel()
-	// Token validity should be checked before in Scopes method
 	_, claims, err := r.client.DecodeAccessToken(ctx, token, r.realm)
-
-	// Get preferred_username from the token payload as the user
-	user, ok := (*claims)["preferred_username"].(string)
-	if !ok {
-		err = errors.New("cannot parse preferred_username from token")
+	if err != nil {
+		return
```
Comment (jortel) [inline auth/provider.go:133]: Returns the error. Fixes the SEGV.

**Review by aufi — APPROVED** [review]
Thanks for the PR, looks good to me!

**Pull request merged**